### ISSUE TITLE
Build fails on Windows with Visual Studio 2019 and non-English language pack

### ISSUE THREAD

**Issue by albertino80** [libexpat/libexpat#418]
Windows 10 64
(my language is Italian)

```
cd expat
mkdir build
cd build
cmake .. -G "Visual Studio 16 2019" && msbuild /m expat.sln
```

```
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.18363.
-- ===========================================================================
--
-- Configuration
--   Prefix ..................... C:/Program Files (x86)/expat
--   Build type .................
--   Shared libraries ........... ON
--   Static CRT ................. OFF
--   Character type ............. char (UTF-8)
--   Binary postfix .............
--
--   Build documentation ........ OFF
--   Build examples ............. ON
--   Build fuzzers .............. OFF
--   Build tests ................ ON
--   Build tools (xmlwf) ........ ON
--   Build pkg-config file ...... OFF
--   Install files .............. ON
--
--   Features
--     // Advanced options, changes not advised
--     Attributes info .......... OFF
--     Context bytes ............ 1024
--     DTD support .............. ON
--     Large size ............... OFF
--     Minimum size ............. OFF
--     Namespace support ........ ON
--
--   Entropy sources
--     rand_s ................... ON
--
-- ===========================================================================
-- Configuring done
-- Generating done
```

Partial Output:

```
2>Building Custom Rule C:/dev_qt/cpp_libs/lib_src/expat/libexpat-master/expat/CMakeLists.txt
2>cl : warning della riga di comando D9002 : l'opzione sconosciuta '-fno-strict-aliasing' verrà ignorata
2>cl : warning della riga di comando D9002 : l'opzione sconosciuta '-fvisibility=hidden' verrà ignorata
2>xmlparse.c
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(173,1): error C2143: errore di sintassi: ')' mancante prima di '('
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(173,1): error C2091: la funzione restituisce una funzione
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(173,1): error C2143: errore di sintassi: ')' mancante prima di 'stringa'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(173,1): error C2143: errore di sintassi: '{' mancante prima di 'stringa'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(173,1): error C2059: errore di sintassi: 'stringa'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(173,1): error C2059: errore di sintassi: ')'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(188,1): error C2143: errore di sintassi: ')' mancante prima di '('
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(188,1): error C2091: la funzione restituisce una funzione
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(188,1): error C2143: errore di sintassi: ')' mancante prima di 'stringa'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(188,1): error C2143: errore di sintassi: '{' mancante prima di 'stringa'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(188,1): error C2059: errore di sintassi: 'stringa'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(188,1): error C2059: errore di sintassi: ')'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(204,1): error C2143: errore di sintassi: ')' mancante prima di '('
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(204,1): error C2091: la funzione restituisce una funzione
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(204,1): error C2143: errore di sintassi: ')' mancante prima di 'stringa'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(204,1): error C2143: errore di sintassi: '{' mancante prima di 'stringa'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(204,1): error C2059: errore di sintassi: 'stringa'
2>C:\dev_qt\cpp_libs\lib_src\expat\libexpat-master\expat\lib\expat.h(204,1): error C2059: errore di sintassi: ')'
```
And many other errros like this

**Comment by hartwork** [comment]
Hi Alberto,

nice to meet you! The AppVeyor CI we have covers up to Visual Studio 2017 so I'm assuming we have something specific to version 2019 here. I'll try to extend the CI to cover 2019. I cannot read Italian error messages so if you can provide any insights on what's going on and ideally replace the output with an English version, that would be appreciated, thank you!

Best, Sebastian

**Comment by albertino80** [comment]
I have investigated a bit.

My compiler version (latest):
`The C compiler identification is MSVC 19.27.29111.0`

I think the problem with MSVC 2019  is when CMake perform tests:
```
Performing Test FLAG_NO_STRICT_ALIASING - Success
Performing Test FLAG_VISIBILITY - Success
```

With older MSVC this reports:
```
Performing Test FLAG_NO_STRICT_ALIASING - Failed
Performing Test FLAG_VISIBILITY - Failed
```

In order to compile successfully I modified `expat\ConfigureChecks.cmake` at the end adding a condition:
```
if(NOT MSVC)
	check_c_compiler_flag("-fno-strict-aliasing" FLAG_NO_STRICT_ALIASING)
	check_c_compiler_flag("-fvisibility=hidden" FLAG_VISIBILITY)
endif()
```

With this, it compiles and run successfully

**Comment by hartwork** [comment]
That's a great hint, thanks!

**Comment by hartwork** [comment]
@albertino80 I got AppVeyor CI to run with the same `MSVC 19.27.29111.0` now and [compilation succeeds unpatched](https://ci.appveyor.com/project/libexpat/libexpat/builds/35393053/job/h2v166dsva955etp), so I'm having trouble reproducing the issue. Any chance that you have Clang installed as well? Any ideas why it fails for you but not on AppVeyor CI? For the related changes please see pull request #420.

**Comment by hartwork** [comment]
@albertino80 any news? I don't feel comfortable with blind fixing things that I cannot reproduce so you help with making this issue reproducible would make an actual difference here. Especially, if this is something you want to have fixed before the upcoming release of Expat 2.2.10 (#415). Thanks in advance!

**Comment by albertino80** [comment]
Ok, I found the problem.

TLDR: after installation of vs2019 English language pack all is fine.

My version of vs2019 works with Italian language pack.

CMake make assertions based on messages produced by the compiler  see `cmake-3.18.3-win64-x64\share\cmake-3.18\Modules\CMakeCheckCompilerFlagCommonPatterns.cmake`

```
macro (CHECK_COMPILER_FLAG_COMMON_PATTERNS _VAR)
  set(${_VAR}
... TRUNCATED ...
    FAIL_REGEX "warning D9002"                             # MSVC, any lang
... TRUNCATED ...
```

When performing the check `check_c_compiler_flag("-fno-strict-aliasing" FLAG_NO_STRICT_ALIASING)` in `CMakeOutput.log` I read (Italian Output):
```
cl : warning della riga di comando D9002 : l'opzione sconosciuta '-fno-strict-aliasing' verrà ignorata
```
The problem is that `FAIL_REGEX` is not matched, and in console: 
```
Performing Test FLAG_NO_STRICT_ALIASING - Success
Performing Test FLAG_VISIBILITY - Success
```

To solve I Installed English language pack for vs2019, now in CMakeError.log I read:
```
cl : command line warning D9002: ignoring unknown option '-fno-strict-aliasing' 
```

And this match `FAIL_REGEX`. So after this step I got in console:
```
-- Performing Test FLAG_NO_STRICT_ALIASING - Failed
-- Performing Test FLAG_VISIBILITY - Failed
```

And everything builds fine.

**Comment by hartwork** [comment]
@albertino80 thanks for a lot for researching and explaining the problem! :pray: 

If I understand correctly, there is no fix needed inside of Expat so I am closing the issue.

**Comment by albertino80** [comment]
Yes, no fix needed.
